pkg/network/http/ebpf_gotls.go

```diff
@@ -178,62 +209,48 @@ func (p *GoTLSProgram) addInspectionResultToMap(result *bininspect.Result, binPa
 
 func (p *GoTLSProgram) attachHooks(result *bininspect.Result, binPath string) error {
 	uid := getUID(binPath)
-	for i, offset := range result.Functions[bininspect.ReadGoTLSFunc].ReturnLocations {
-		probeID := manager.ProbeIdentificationPair{
-			EBPFSection:  readReturnProbe,
-			EBPFFuncName: readReturnFuncName,
-			UID:          makeReturnUID(uid, i),
-		}
-		err := p.manager.AddHook("", &manager.Probe{
-			ProbeIdentificationPair: probeID,
-			BinaryPath:              binPath,
-			// Each return probe needs to have a unique uid value,
-			// so add the index to the binary UID to make an overall UID.
-			UprobeOffset: offset,
-		})
-		if err != nil {
-			return fmt.Errorf("could not add hook to read return in offset %d due to: %w", offset, err)
+
+	for function, uprobes := range functionToProbes {
+		if functionsConfig[function].IncludeReturnLocations && uprobes.returnInfo != nil {
```

Comment:
if `functionsConfig[function].IncludeReturnLocations && uprobes.returnInfo == nil` we should report an error